lib/spl/src/unk_0209DD54.c

```diff
@@ -0,0 +1,1029 @@
+
+    if (alpha == 0) {
+        return;
+    }
+
+    sclY = ptcl->unk_30;
+    sclX = FX_MUL(sclY, aspect);
+
+    switch (mgr->unk_40.unk_00->p_res->unk_00->unk_48.unk_07_4) {
+    case 0:
+        sclX = FX_MUL(sclX, ptcl->unk_34);
+        sclY = FX_MUL(sclY, ptcl->unk_34);
+        break;
+
+    case 1:
+        sclX = FX_MUL(sclX, ptcl->unk_34);
+        break;
+
+    case 2:
+        sclY = FX_MUL(sclY, ptcl->unk_34);
+        break;
+    }
+
+    if (!mgr->unk_40.unk_00->p_res->unk_00->unk_00.unk_06_7) {
+        trs.x = ptcl->unk_08.x + ptcl->unk_38.x;
+        trs.y = ptcl->unk_08.y + ptcl->unk_38.y;
+        trs.z = ptcl->unk_08.z + ptcl->unk_38.z;
+
+        dir = ptcl->unk_14;
+
+        look.x = cmr->_02;
+        look.y = cmr->_12;
+        look.z = cmr->_22;
+
+        VEC_CrossProduct(&dir, &look, &dir);
+        if (dir.x == 0 && dir.y == 0 && dir.z == 0) {
+            return;
+        }
+
+        VEC_Normalize(&dir, &dir);
+        MI_Copy36B(cmr, &mtx);
+        MTX_MultVec33(&dir, &mtx, &dir);
+        MTX_MultVec43(&trs, cmr, &trs);
+
+        vel_dir = ptcl->unk_14;
+        VEC_Normalize(&vel_dir, &vel_dir);
+
+        dot = FX_MUL(vel_dir.x, -cmr->_02) + FX_MUL(vel_dir.y, -cmr->_12) + FX_MUL(vel_dir.z, -cmr->_22);
+        if (dot < 0) {
+            dot = -dot;
+        }
+
+        dot = FX_MUL(sclY, FX_MUL(FX32_ONE - dot, (fx32)mgr->unk_40.unk_00->p_res->unk_00->unk_48.unk_05_0) + FX32_ONE);
+        load._00 = FX_MUL(dir.x, sclX);
+        load._10 = FX_MUL(-dir.y, dot);
+        load._30 = trs.x;
+        load._31 = trs.y;
+        load._01 = FX_MUL(dir.y, sclX);
+        load._11 = FX_MUL(dir.x, dot);
+        load._20 = 0;
+        load._21 = 0;
+        load._02 = 0;
+        load._12 = 0;
+        load._22 = FX32_ONE;
+        load._32 = trs.z;
+
+        G3_Identity();
+        G3_MultMtx43(&load);
+    } else {
+        trs.x = ptcl->unk_08.x + ptcl->unk_38.x - mgr->unk_40.unk_00->p_res->unk_00->unk_04.x;
+        trs.y = ptcl->unk_08.y + ptcl->unk_38.y - mgr->unk_40.unk_00->p_res->unk_00->unk_04.y;
+        trs.z = ptcl->unk_08.z + ptcl->unk_38.z - mgr->unk_40.unk_00->p_res->unk_00->unk_04.z;
+
+        dir = ptcl->unk_14;
+
+        look.x = cmr->_02;
+        look.y = cmr->_12;
+        look.z = cmr->_22;
+
+        VEC_CrossProduct(&dir, &look, &dir);
+        if (dir.x == 0 && dir.y == 0 && dir.z == 0) {
+            return;
+        }
+
+        VEC_Normalize(&dir, &dir);
+        MI_Copy36B(cmr, &mtx);
+        MTX_MultVec33(&dir, &mtx, &dir);
+        MTX_MultVec43(&trs, cmr, &trs);
+
+        vel_dir = ptcl->unk_14;
+        VEC_Normalize(&vel_dir, &vel_dir);
+
+        dot = FX_MUL(vel_dir.x, -cmr->_02) + FX_MUL(vel_dir.y, -cmr->_12) + FX_MUL(vel_dir.z, -cmr->_22);
+        if (dot < 0) {
+            dot = -dot;
+        }
+
+        dot = FX_MUL(sclY, FX_MUL(FX32_ONE - dot, (fx32)mgr->unk_40.unk_00->p_res->unk_00->unk_48.unk_05_0) + FX32_ONE);
+        load._00 = FX_MUL(dir.x, sclX);
```

Comment:
does it match if you use the `.m` accessor?

Reply:
Yeah